### ISSUE TITLE
Tail plugin can't handle UTF-16 files

### ISSUE THREAD

**Issue by SolracLeinad** [fluent/fluent-bit#7497]
## Bug Report

**Describe the bug**
Fluent Bit will add an extra space to each character readed from any UTF-16 file.

**To Reproduce**
- Set your output to stdout
- Take any working log file and convert it to UTF-16
- Process that log again.
- Profit

- Log message (expected):
```
{"log":"YOUR LOG MESSAGE HERE","stream":"stdout","time":"2018-06-11T14:37:30.681701731Z"}
```

- What you actually get:
```
{"log":"2 0 1 8 - 0 6 - 1 1 T 1 4 : 3 7 : 3 0 . 6 8 1 7 0 1 7 3 1 Z  Y O U R   L O G   M E S S A G E   H E R E"}
```

**Expected behavior**
I expect to get a normal message.
{"log":"YOUR LOG MESSAGE HERE","stream":"stdout","time":"2018-06-11T14:37:30.681701731Z"}

**Screenshots**
This is an UTF-16 log file digested by fluent bit:
![image](https://github.com/fluent/fluent-bit/assets/23008975/071b2847-6de4-4e51-815e-2aeee0832712)

It should look like this:
![image](https://github.com/fluent/fluent-bit/assets/23008975/f0c9914d-73df-4a95-86b1-04f61349f445)

**Your Environment**
* Version used: fluent-bit-2.1.4-win64
* Configuration: 

[INPUT]
    Name         tail
    path         Logfile.txt
    Read_from_Head true

[OUTPUT]
    name  stdout
    Match *

* Environment name and version (e.g. Kubernetes? What version?):
* Server type and version: Windows server 2012
* Operating System and version: Windows server 2012
* Filters and plugins: None

**Additional context**
Im just trying to read my log file but it's adding extra spaces in between chars.

**Comment by leonardo-albertovich** [comment]
Hi @SolracLeinad, this is something we intend to improve but don't have an ETA or roadmap at the moment so we'll leave this issue open.

The only suggestion I can make at the moment trying to use a lua or wasm filter script to convert the data from utf-16 to utf-8 before applying any other filters or parsers.

**Comment by SolracLeinad** [comment]
Thanks @leonardo-albertovich 

I decided to use an "universal parser":  `Regex ^(?<log>\w.*)$`  to get each line sent to a processor. On the processor side I'm stripping the extra spaces and everything was fine until a new problem arose.

The Tail plugin is now ignoring any changes in the UTF-16 log file.

My debug logs look like this now:
![image](https://github.com/fluent/fluent-bit/assets/23008975/6643d11b-89cd-4dd4-b441-4aca2bb1f454)

And this:
![image](https://github.com/fluent/fluent-bit/assets/23008975/016f8ceb-c735-483d-893e-44da0dee41bc)

I tried to use `Inotify_Watcher false` but it's not supported by the windows version:
`[error] [config] tail: unknown configuration property 'inotify_watcher'. The following properties are allowed: path...`


I think I'm gonna have to wait for the UTF-16 support, as for now I can't use Fluent Bit.
